[WIP]: Updated Gen.AI and NNCF documentation (#22793)
openvinotoolkit/openvino: docs/articles_en/openvino-workflow/model-optimization-guide/weight-compression.rst

    * ``nncf.SensitivityMetric.MEAN_ACTIVATION_MAGNITUDE`` - requires dataset. The mean magnitude of the layers' inputs multiplied by inverted 8-bit quantization noise.

  * ``all_layers`` - boolean parameter that enables INT4 weight quantization of all Fully-Connected and Embedding layers including first and last layer in the model. 

>   * ``awq`` - boolean parameter that enables AWQ method for more accurate INT4 weight quantization. Especially helpful when weights of all the layers quantized to 4 bits. The method is not friendly to Dynamic Quantization of activations. Requires dataset.

Author: @tsavina, `The method is not compatible with Dynamic Quantization of activations.` is not the right statement. It is fully compatible but can lead to worse accuracy sometimes.

         :language: python
         :fragment: [compression_4bit]

For data-aware weight compression refer to the following `example <https://github.com/openvinotoolkit/nncf/tree/develop/examples/llm_compression/openvino/tiny_llama>`__.

.. note::

   OpenVINO also supports 4-bit models from Hugging Face `Transformers <https://github.com/huggingface/transformers>`__ library optimized 
   with `GPTQ <https://github.com/PanQiWei/AutoGPTQ>`__. In this case, there is no need for an additional model optimization step because model conversion will automatically preserve the INT4 optimization results, allowing model inference to benefit from it.

The table below shows examples of text-generation Language Models with different optimization settings in a data-free setup when no dataset is used at optimization step.

> The Perplexity metric is measured on `Lambada OpenAI dataset <https://github.com/openai/gpt-2/issues/131#issuecomment-497136199>`__.

Reviewer: ```suggestion
The Perplexity metric is measured on the `Lambada OpenAI dataset <https://github.com/openai/gpt-2/issues/131#issuecomment-497136199>`__.
```

     - INT4_SYM,group_size=64,ratio=0.8
     - 2.98
     - 8.0

> The following table shows accuracy metric in a data-aware 4-bit weight quantization setup measured on `Wikitext dataset <https://arxiv.org/pdf/1609.07843.pdf>`__.

Reviewer: ```suggestion
The following table shows accuracy metric in a data-aware 4-bit weight quantization setup measured on the `Wikitext dataset <https://arxiv.org/pdf/1609.07843.pdf>`__.
```

.. list-table::
   :widths: 40 55 25 25
   :header-rows: 1

   * - Model
     - Optimization
     - Word perplexity\*
     - Model Size (Gb)
   * - meta-llama/llama-7b-chat-hf
     - FP32
     - 11.57
     - 12.61
   * - meta-llama/llama-7b-chat-hf
     - INT4_SYM,group_size=128,ratio=1.0,awq=True
     - 12.34
     - 2.6
   * - stabilityai_stablelm-3b-4e1t
     - FP32
     - 10.17
     - 10.41
   * - stabilityai_stablelm-3b-4e1t
     - INT4_SYM,group_size=64,ratio=1.0,awq=True
     - 10.89
     - 2.6
   * - HuggingFaceH4/zephyr-7b-beta
     - FP32
     - 9.82
     - 13.99
   * - HuggingFaceH4/zephyr-7b-beta
     - INT4_SYM,group_size=128,ratio=1.0
     - 10.32
     - 2.6

> \*Perplexity metric in both tables was measured without Dynamic Quantization feature enabled in the OpenVINO runtime.

Reviewer: ```suggestion
\*Perplexity metric in both tables was measured without the Dynamic Quantization feature enabled in the OpenVINO runtime.
```

   

Auto-tuning of Weight Compression Parameters
############################################

> The important question that may arise is how to find a configuration of weight compression parameters that is best suited to a particular model.
> We provide an `example <https://github.com/openvinotoolkit/nncf/tree/develop/examples/llm_compression/openvino/tiny_llama_find_hyperparams>`__ 
> for that matter where the weight compression parameters are being searched from the subset of values. To speed up the search we use self-designed 
> validation pipeline that we called `WhoWhatBench <https://github.com/openvinotoolkit/openvino.genai/tree/master/llm_bench/python/who_what_benchmark>`__ 
> that can quickly evaluate the changes in accuracy of the optimized model compared to the baseline.

Reviewer: ```suggestion
To find the optimal weight compression parameters for a particular model, refer to the `example <https://github.com/openvinotoolkit/nncf/tree/develop/examples/llm_compression/openvino/tiny_llama_find_hyperparams>`__ , where weight compression parameters are being searched from the subset of values. To speed up the search, a self-designed 
validation pipeline called `WhoWhatBench <https://github.com/openvinotoolkit/openvino.genai/tree/master/llm_bench/python/who_what_benchmark>`__ is used. 
The pipeline can quickly evaluate the changes in the accuracy of the optimized model compared to the baseline.
```
